ndk/src/main/jni/handlers/signal_handler.c

```diff
@@ -161,7 +161,10 @@ void bsg_invoke_previous_signal_handler(int signum, siginfo_t *info,
 
 void bsg_handle_signal(int signum, siginfo_t *info,
                        void *user_context) __asyncsafe {
-  if (bsg_global_env == NULL || bsg_global_env->handling_crash) {
+  if (bsg_global_env == NULL) {
```

Comment:
I'm not sure I fully understand the distinction here, what was the issue with the previous code?

Reply:
The following line didn't make sense if `env` is null and was a potential npe.

Reply:
Ah, that makes sense now! I initially thought the first condition checked `bsg_global_env != NULL`.